Major cleanup

jdigger opened the pull request:
A lot of cleanup. From a CR perspective, it will be much easier to view each individual commit.

jdigger committed:
Upgraded to gradle-plugin-cq-bundle v2.1.0

Brought over a few functions that used to be in -bundle v.2.0 and put them in CqPackageHelper such as
  * doAcrossServers
  * startInactiveBundles
  * validateBundles
  * validateRemoteBundles
  * validateAllBundles
  * hasAnInactiveBundle
  * uninstallBundles
  * uninstallAllBundles

Removed the "checkBundleStatus" since it's redundant with "validateBundles"

Changed a number of methods in CqPackageHelper to return their status instead of 'void'. Still need to change CqPackagePlugin and the like to check for the status.

jdigger committed:
Added true integration/functional testing

CqPackagePluginSpec has been expanded to create an embedded instance of Jetty for functional testing. With that, have added a LOT more tests.

CqPackagePlugin now actually does apply the "base" plugin, conforming to its documentation.

Removed the "reinstall" task since it's just a convenience task tying other tasks together. Projects using this plugin can do that better for themselves.

Updated the .travis.yml to use the JDK 1.8 compiler. Even though JDK 1.7 byte code is generated for the plugin, the Jetty libraries require JDK 1.8. Since it's only for test runtime, it doesn't create a JDK 1.8 runtime dependency for the plugin.

jdigger committed:
Better support for "package" system property for "upload" package task

If the "package" system property is not specified and there is not output from "createPackage", then it fails fast with a clear error message. Fixes GH-15

Added the "remove" task as a dependency for "upload" to prevent unexpected overlays.

jdigger committed:
RuntimePackageProperties.packageProperties now wants a PackageId

"packageProperties" accepts a Jackrabbit PackageId instead of just a String (the package name) in preparation for implementing GH-8

Changed PackageHelperSpec to extend from ProjectSpec instead of just Specification to let Netflix's Nebula handle more of the boilerplate for testing

Removed deprecated CqPackageHelper.getPackageInfo()

jdigger committed:
Added more tests to CqPackageHelperSpec

jdigger committed:
Fixed model declaration for the "createPackage" task

If the "jar" task is added to the project, "createPackage" is declared that it must run after it.

The generated .zip is added to Gradle's artifact publication set so that things like "assemble" know about it.

jdigger committed:
A lot of cleanup to the task definitions

Renamed the tasks to be less generic (e.g., "upload" -> "uploadPackage")

Removed tasks that are redundant or are of marginal utility.

Added stronger typing to task dependencies in plugin declaration, as well as more javadocs

Changed the actual task declarations to be consistent in the form of declaration.

Fixes GH-13

jdigger committed:
Updated the plugin documentation

jdigger commented:
For a good top-level understanding of what this does, see the greatly updated and expanded [plugin documentation](../blob/refactor/docs/CqPackagePlugin.adoc).

jdigger commented:
Please review, @sikarwarvimal @jbornemann

/cc @itgumby @cculb @jscelza @hsfear @fortman @hsfear

jbornemann commented on .travis.yml:
```diff
@@ -2,4 +2,12 @@ language: java
 script: TERM=dumb ./gradlew clean check
 jdk:
-    - openjdk7
+    - oraclejdk8
```
Should we update build compatibility in build.gradle to reflect this?

jdigger replied:
What do you mean? For the byte code generation? JDK 1.8 is for test runtime only. See commit message for 9415a2f0ccd4cff6641d8fe15d9f72e60fd0fba6

jbornemann replied:
FWIW, when building with 1.7, javaDoc task fails : 

```
:javadoc
javadoc: error - invalid flag: -Xdoclint:none
```

Test compilation fails as well : 

```
* What went wrong:
Execution failed for task ':compileTestGroovy'.
> com/twcable/gradle/sling/SlingServerFixture : Unsupported major.minor version 52.0
```

sikarwarvimal commented:
@jdigger i couldn't find `./gradlew install` task. Is it not there anymore?

jdigger commented:
"install" is not there any more, as part of GH-13. See the greatly updated and expanded [plugin documentation](../blob/refactor/docs/CqPackagePlugin.adoc).

ItGumby commented:
liking the plugin docs.

jdigger committed:
'createPackage' explicitly depends on any tasks that create bundles

All tasks inside the project as well as in projects that it depends on via the 'cq_package' configuration are included

Fixes GH-2
